Modules/gcmodule.c

```diff
@@ -227,20 +267,38 @@ append_objects(PyObject *py_list, PyGC_Head *gc_list)
     return 0;
 }
 
+#if GC_DEBUG
+static void
+validate_list(PyGC_Head *head, uintptr_t expected_mask)
```

Comment:
@pitrou @serhiy-storchaka 
How do you think this function?  Should I remove these checks and GC_DEBUG flag?

This validation was useful while hacking, and I think it's good document about when MASK flags are set and cleared.
But I can replace them to just a line of comment, of course.

Reply:
I'm ok with this function. Have you tried to always enable it in debug mode, or is it too slow?